lib/rubocop/cop/sorbet/sigils/enforce_sigil_order.rb

```diff
@@ -49,6 +51,13 @@ def autocorrect(_node)
             tokens.each_with_index do |token, index|
               corrector.replace(token.pos, expected[index].text)
             end
+
+            # Remove blank lines between the magic comments
+            lines = Set.new
+            tokens.each { |token| lines << token.line }
+            each_extra_empty_line(lines.sort) do |range|
+              corrector.remove(range)
+            end
```

Comment:
Can we not just `lines.each` and in the block just remove the source range for a line if it is `empty?` I am not sure if I understand what the double loop in the `each_extra_empty_line` is doing.

Reply:
I'm in the process of simplifying what I copy pasted over 😄  
I will try that too.

Reply:
The `lines` does not represent the lines in the file. It only consists of lines that are not empty. For example if we have 
```
1 # frozen_string_literal: true
2
3
4 # typed: true
```
`lines` contains `[1, 4]`

Therefore, in order to check for emptiness of lines `2, 3` we need to keep track of previous and current lines. The inner loop is iterating between "the first potentially empty line" and "current line". 

In our example we'd have only 1 iteration and these would be "2" and "4" respectively. Since it's iterating between a range of these, `2..4`, it's iterating over both of the potentially empty lines and checks for their emptiness.

Lmk if that example helps.

Reply:
Yes, but that still feels like a lot of overhead. I have a suggestion for a simpler implementation. Will resolve this and comment that.